coveo-settings/coveo_settings/setting_abc.py

```diff
@@ -263,3 +280,78 @@ def __contains__(self, item: Any) -> bool:
         """Tells if item is in `value`. Will raise on unsupported types or missing values."""
         self._raise_if_missing()
         return item in self.value  # type: ignore[operator]
+
+
+AdapterHandler = Callable[[str], Optional[ConfigValue]]
+
+
+class _SchemeDispatch:
+    """Decorator that registers a function as a callback for a specific scheme."""
+
+    matchers: Final[Dict[Pattern, AdapterHandler]] = {}
+
+    def __init__(self, scheme: str) -> None:
+        """
+        Reasonable schemes will use the same delimiter pattern within an application for consistency and will
+        contain several characters for uniqueness.
+
+        Examples:
+            - ssm://
+            - s3->
+            - {api}
+        """
+        self.matcher = re.compile(rf"^{re.escape(scheme)}(?P<resource>.+)$", flags=re.IGNORECASE)

```

Comment:
Not sure a full fledged regex was necessary here.

Reply:
It was useful at some moment when I prototyped the feature, but it evolved and it's not doing much anymore.